dpctl-capi/source/dpctl_sycl_device_interface.cpp

```diff
@@ -387,3 +389,95 @@ bool DPCTLDevice_HasAspect(__dpctl_keep const DPCTLSyclDeviceRef DRef,
     }
     return hasAspect;
 }
+
+__dpctl_give DPCTLDeviceVectorRef
+DPCTLDevice_CreateSubDevicesEqually(__dpctl_keep const DPCTLSyclDeviceRef DRef,
+                                    size_t count)
+{
+    // vector_class<DPCTLSyclDeviceRef> *Devices = nullptr;
+    DPCTLDeviceVectorRef Devices = nullptr;
+    auto D = unwrap(DRef);
+    if (D) {
+        try {
+            auto subDevices = D->create_sub_devices<
+                info::partition_property::partition_equally>(count);
+            // Devices = new vector_class<DPCTLSyclDeviceRef>();
+            Devices = DPCTLDeviceVector_Create();
+            DPCTLDeviceVector_Resize(Devices, subDevices.size());
+            for (int i = 0; i < subDevices.size(); i++) {
+                // auto &sd : subDevices
+                DPCTLDeviceVector_SetAt(Devices, i, subDevices[i]);
```

Comment:
You will need `DPCTLDeviceVector_SetAt(Devices, i, wrap(new device(subDevices[i])));`